src/Components/CreateImageWizard/steps/Kernel/components/KernelArguments.tsx

```diff
@@ -2,8 +2,57 @@ import React from 'react';
 
 import { FormGroup } from '@patternfly/react-core';
 
+import { useAppSelector } from '../../../../../store/hooks';
+import { useGetOscapCustomizationsQuery } from '../../../../../store/imageBuilderApi';
+import {
+  addKernelArg,
+  removeKernelArg,
+  selectComplianceProfileID,
+  selectDistribution,
+  selectKernel,
+} from '../../../../../store/wizardSlice';
+import ChippingInput from '../../../ChippingInput';
+import { isKernelArgumentValid } from '../../../validators';
+
 const KernelArguments = () => {
-  return <FormGroup isRequired={false} label="Append"></FormGroup>;
+  const kernelAppend = useAppSelector(selectKernel).append;
+
+  const release = useAppSelector(selectDistribution);
+  const complianceProfileID = useAppSelector(selectComplianceProfileID);
+
+  const { data: oscapProfileInfo } = useGetOscapCustomizationsQuery(
+    {
+      distribution: release,
+      // @ts-ignore if complianceProfileID is undefined the query is going to get skipped, so it's safe here to ignore the linter here
+      profile: complianceProfileID,
+    },
+    {
+      skip: !complianceProfileID,
+    }
+  );
+
+  const requiredByOpenSCAP = kernelAppend.filter((arg) =>
+    oscapProfileInfo?.kernel?.append?.split(' ').includes(arg)
+  );
+
+  const notRequiredByOpenSCAP = kernelAppend.filter(
```

Comment:
Nitpick: could you not just have `!requiredByOpenSCAP`?

Reply:
Ah right, the code was a bit redundant :sweat_smile: thank you! Added a new commit that should fix that. Also found one small bug so added a fix as well.